### PR TITLE
Trivial refactorings around generator document identity

### PR DESCRIPTION
**Pull request by jasonmalinowski**
Just refactoring some types.

### CODE REVIEW

File: src/VisualStudio/Core/Impl/SolutionExplorer/DiagnosticItem/SourceGeneratorItem.cs [hunk 1]
```diff
@@ -13,18 +13,18 @@ internal sealed partial class SourceGeneratorItem : BaseItem
     {
         public ProjectId ProjectId { get; }
 
-        public string GeneratorAssemblyName { get; }
+        public SourceGeneratorIdentity Identity { get; }
 
         // Since the type name is also used for the display text, we can just reuse that. We'll still have an explicit
         // property so the assembly name/type name pair that is used in other places is also used here.
         public string GeneratorTypeName => base.Text;
```
Comment (ryzngard) [inline src/VisualStudio/Core/Impl/SolutionExplorer/DiagnosticItem/SourceGeneratorItem.cs:20]: Is this still used?

**Review by ryzngard — APPROVED** [review]
Overall LGTM

File: src/VisualStudio/Core/Impl/SolutionExplorer/DiagnosticItem/SourceGeneratorItem.cs (hunk 1 above)
Comment (jasonmalinowski) [reply src/VisualStudio/Core/Impl/SolutionExplorer/DiagnosticItem/SourceGeneratorItem.cs:20]: Oh probably not, great catch.

Comment (jasonmalinowski) [reply src/VisualStudio/Core/Impl/SolutionExplorer/DiagnosticItem/SourceGeneratorItem.cs]: Fixed.

Comment (jasonmalinowski) [reply src/VisualStudio/Core/Impl/SolutionExplorer/DiagnosticItem/SourceGeneratorItem.cs]: Turns out it _was_ used, didn't realize there was a partial there. Fixed that too.

**Pull request merged**